src/lib/storage/frame_of_reference_segment.hpp

```diff
@@ -56,7 +56,15 @@ class FrameOfReferenceSegment : public BaseEncodedSegment {
 
   const AllTypeVariant operator[](const ChunkOffset chunk_offset) const final;
 
-  const std::optional<T> get_typed_value(const ChunkOffset chunk_offset) const;
+  const std::optional<T> get_typed_value(const ChunkOffset chunk_offset) const {
+    // not in cpp for performance reasons
```

Comment:
Why does it get slower in the cpp?

Reply:
Because it could not be inlined quite as well.

Reply:
Is there an explanation for it or is it only a phenomenon we observed? 

Also, `performance reasons` makes me wonder which reasons it could have. `inlining works better` does not leave a doubt/question.

Reply:
> Is there an explanation for it or is it only a phenomenon we observed?

Well, the compiler cannot inline something it cannot see.

> Also, performance reasons makes me wonder which reasons it could have. inlining works better does not leave a doubt/question.

done